build/replace/Jenkinsfile

```diff
@@ -0,0 +1,50 @@
+pipeline {
+  agent {
+    dockerfile {
+      dir 'build'
+    }
+  }
+
+  options {
+    buildDiscarder(logRotator(numToKeepStr: '10'))
+    disableConcurrentBuilds()
+  }
+
+  triggers {
+    cron '@midnight'
+  }
+  
+  parameters {
+    booleanParam name: 'dryRun', defaultValue: true, description: 'whether the build should push the changes or not'
+    string(name: 'sedRegexp', defaultValue: '', description: 'The sed replacement: s#search#replace#g')
+    string(name: 'fileSelector', defaultValue: '', description: 'Bash file selector e.g. folder/**/File.*')
+    string(name: 'sourceBranch', defaultValue: 'master', description: 'The branch to apply this change on e.g. master, release/11.1, release/10.0')
+    string(name: 'newBranch', defaultValue: 'replace', description: 'The name of the new branch, which gets pushed if it is not a dry run')
+    string(name: 'commitMessage', defaultValue: '', description: 'The commit message')
+  }
+
+  stages {
+    stage('replace') {
+      when {
+        expression { !params.sedRegexp.isEmpty() && !params.fileSelector.isEmpty() }
```

Comment:
maybe those two should have a default value. now the pipeline will never run even in dryRun mode right?